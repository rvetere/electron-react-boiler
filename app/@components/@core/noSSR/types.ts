import { ReactNode } from 'react'

export interface INoSSRProps {
  onSSR?: ReactNode
  renderOnServerForBots?: boolean
}

export interface INoSSRState {
  mounted: boolean
}
