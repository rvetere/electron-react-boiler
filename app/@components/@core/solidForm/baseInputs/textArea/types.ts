import { IValidationDisplayConfig } from '@core/solidForm/elements/infoSlot/validationDisplay/types'
import { IFormControllerInputProps } from '@core/solidForm/formController/types'
import { IInputDefaultState } from '@core/solidForm/types'
import { ValidatorFn } from '@core/solidForm/validators/types'
import React, { ReactNode } from 'react'

export interface ITextAreaProps extends React.InputHTMLAttributes<HTMLTextAreaElement> {
  name: string
  inputProps: IFormControllerInputProps
  validators?: ValidatorFn[]
  inverted?: boolean
  label?: ReactNode
  labelAppendix?: ReactNode
  hideValidation?: boolean
  infoSlot?: (validationConfig: IValidationDisplayConfig) => ReactNode
}

export type ITextAreaState = IInputDefaultState
