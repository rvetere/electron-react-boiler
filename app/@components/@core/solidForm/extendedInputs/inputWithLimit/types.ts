import { IInputTextProps } from '@core/solidForm/baseInputs/input/types'
import { ITextAreaProps } from '@core/solidForm/baseInputs/textArea/types'
import { FieldWithLimitType } from '.'

export interface IInputTextWithLimitProps extends IInputTextProps {
  limit: number
  type?: FieldWithLimitType
}

export interface ITextAreaWithLimitProps extends ITextAreaProps {
  limit: number
  type?: FieldWithLimitType
}

export interface IInputWithLimitState {
  isFocused: boolean
  valueLength: number
}
