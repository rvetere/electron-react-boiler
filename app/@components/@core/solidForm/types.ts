import { InputValue } from '.'
import { IValidationMessage } from './elements/infoSlot/validationDisplay/types'

export enum LabelPosition {
  LabelLeft = 'LabelLeft',
  LabelRight = 'LabelRight'
}

export interface IInputDefaultState {
  value?: InputValue
  isValid: boolean
  isTouched: boolean
  validationMessages: IValidationMessage[]
}
