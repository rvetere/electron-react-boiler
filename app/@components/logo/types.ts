export interface ILogoSmallProps {
  sticker?: boolean
}
