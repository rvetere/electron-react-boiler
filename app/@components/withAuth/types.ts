import { IFormValues } from '@core/solidForm'
import { IAuthState } from '@reducers/types'
import { ReactNode } from 'react'
import { RouterProps } from 'react-router'

export interface IWithAuthInnerControllerProps {
  children: (auth: IAuthState) => ReactNode
  auth: IAuthState
  signIn: (authState: IAuthState) => void
}

export interface IWithAuthViewProps {
  onSubmit: (values: IFormValues) => void
}

export interface ITokenData {
  name: string
  email: string
  roles: string[]
}

export interface IPrivateRouteProps extends RouterProps {
  component: React.SFC<RouterProps>
  auth: IAuthState
  adminOnly?: boolean
}
