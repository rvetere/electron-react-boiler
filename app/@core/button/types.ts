import { SyntheticEvent } from 'react'

// TODO ...props sollte entfernt werden und die erlaubten Eigenschaften explizit hier aufgefÃ¼hrt werden
export interface IButtonProps {
  type?: string
  loading?: boolean
  primary?: boolean
  secondary?: boolean
  small?: boolean
  outline?: boolean
  big?: boolean
  inverted?: boolean
  block?: boolean
  mobileNoBlock?: boolean
  fullSize?: boolean
  disabled?: boolean
  title?: string
  tabIndex?: number
  onClick?: (event: SyntheticEvent) => void
  onMouseDown?: (event: SyntheticEvent) => void
  onMouseEnter?: () => void
  onMouseLeave?: () => void
}

export interface IButtonStylings {
  className: string
  otherProps: object
}
