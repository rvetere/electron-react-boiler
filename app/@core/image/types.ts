import { ImageSetItem } from '@graphql/types/generated'
import { ReactNode } from 'react'

export interface IImageProps {
  src: string
  alt: string
  loadingSpinner?: ReactNode
  width?: number
  height?: number
  imageSet?: Array<ImageSetItem | null>
  showImageSet?: boolean
  renderDimensionsAttributes?: boolean
  onLoad?: () => void
}

export interface IImageControllerState {
  isVisible: boolean
  width: number
  height: number
}

export interface IImageViewProps {
  isVisible: boolean
  src: string
  width: number
  height: number
  alt: string
  renderDimensionsAttributes: boolean
  onLoad?: () => void
}

export interface IImageControllerRenderProps {
  width: number
  height: number
  isVisible: boolean
  src: string
  onLoad: () => void
  onIntersection: (ev: IntersectionObserverEntry, unobserve: () => void) => void
}

export interface IImageControllerProps {
  width?: number
  height?: number
  src: string
  onLoad?: () => void
  children: (data: IImageControllerRenderProps) => ReactNode
}

export interface IImageListViewProps {
  imageSet: Array<ImageSetItem | null>
  width: number
  height: number
  isVisible: boolean
  renderDimensionsAttributes: boolean
  animationStarted: boolean
  activeImageIndex: number
  startAnimation: () => void
  endAnimation: () => void
  onLoad?: () => void
}

export interface IImageListControllerRenderProps {
  startAnimation: () => void
  endAnimation: () => void
  animationStarted: boolean
  activeImageIndex: number
}

export interface IImageListControllerProps {
  imageSet: Array<ImageSetItem | null>
  children: (data: IImageListControllerRenderProps) => ReactNode
}

export interface IImageListControllerState {
  animationStarted: boolean
  activeImageIndex: number
}
