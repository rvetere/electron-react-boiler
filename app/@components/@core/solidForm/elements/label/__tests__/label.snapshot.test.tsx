/* eslint-env jest */
import { getByClass } from '@testing/classQueries'
import { cleanup, render, waitForElement } from 'react-testing-library'
import { Label } from '..'
import { ILabelProps } from '../types'

describe('Label', () => {
  afterEach(cleanup)
  const mockProps: ILabelProps = {
    label: 'a label',
    inverted: false
  }

  test('Label component should render correctly', () => {
    const { container } = render(<Label {...mockProps}>label text</Label>)

    expect(container).toMatchSnapshot()
  })

  test('Label should support appendix', async () => {
    const expected = 'addon'
    const { container, getByText } = render(
      <Label {...mockProps} appendix={expected}>
        label text
      </Label>
    )

    const appendix = await waitForElement(() => getByClass(container, 'appendix'))
    expect(appendix).toBeTruthy()

    expect(getByText(expected)).toBeTruthy()
  })
})
