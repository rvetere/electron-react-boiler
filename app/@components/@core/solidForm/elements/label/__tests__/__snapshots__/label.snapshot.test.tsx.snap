// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Label component should render correctly 1`] = `
<div>
  <label
    class="formLabel label"
  >
    <span>
      a label
    </span>
    label text
  </label>
</div>
`;
